### ISSUE TITLE
Whitelist private attributes

### ISSUE THREAD

**Issue by brandon-leapyear** [microsoft/pyright#4267]
**Is your feature request related to a problem? Please describe.**
We override and call `_repr_html_` for integration with IPython, and it would be nice to avoid triggering `reportPrivateUsage` for this.

e.g.
```py
@dataclass
class Parent:
  children: list[Child]
  def _repr_html_(self):
    return "<ul>{}</ul>".format("".join(child._repr_html_() for child in self.children))

class Child:
  def _repr_html_(self):
    return "<li>Child</li>"
```

**Describe the solution you'd like**
Add `allowedPrivateUsage` configuration that takes a list of strings.

**Additional context**

**Comment by erictraut** [comment]
It's unlikely that we will provide a configuration to override such a specific diagnostic check.

If you want to override this diagnostic on a per-file basis, you can add a comment at the top of the file: `# pyright: reportPrivateUsage=false`. Alternatively, you can suppress a diagnostic on a particular line using `# pyright: ignore[reportPrivateUsage]`.
